src/views/ManageList.jsx

```diff
@@ -1,7 +1,118 @@
-export function ManageList() {
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+import toast, { Toaster } from 'react-hot-toast';
+
+export function ManageList({ listPath }) {
+	console.log('listPath:', listPath);
+	const [itemName, setItemName] = useState('');
+	const [itemNextPurchaseTimeline, setItemNextPurchaseTimeline] = useState('');
+
+	const handleItemNameTextChange = (e) => {
+		setItemName(e.target.value);
+	};
+
+	const handleNextPurchaseChange = (e) => {
+		setItemNextPurchaseTimeline(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		console.log('Submitting item:', {
+			itemName,
+			itemNextPurchaseTimeline,
+			listPath,
+		});
+
+		try {
+			await toast.promise(
+				addItem(listPath, {
+					itemName,
+					daysUntilNextPurchase: itemNextPurchaseTimeline,
+				}),
+				{
+					pending: 'Adding item to list.',
+					success: `${itemName} successfully added to your list!`,
+					error: `${itemName} failed to add to your list. Please try again!`,
+				},
+				{
+					style: {
+						minWidth: '250px',
+					},
+				},
+				console.log('Item added to list:', itemName),
+			);
+		} catch (error) {
+			console.error('Failed to add item:', error);
+		}
+	};
+
 	return (
-		<p>
-			Hello from the <code>/manage-list</code> page!
-		</p>
+		<div>
+			<p>
+				Hello from the <code>/manage-list</code> page!
+			</p>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="item-name">
+					Item:
+					<input
+						id="item-name"
+						type="text"
+						name="item"
+						value={itemName}
+						onChange={handleItemNameTextChange}
+						required
+						aria-label="Enter the item name"
+					/>
+				</label>
+				<br />
+				<p>
+					When to buy: <br />
```

Comment:
did we consider using a `fieldset` and `legend` here? The `fieldset` is a nice way of grouping fields that are alike, and the `legend` allows you to give that fieldset a "caption." It would result in a very similar result but would have the added benefit of following html semantics, which can go a long way for screen readers :)

Reply:
Thank you for this! It was exactly what we were going for but couldn't find the right tags!